File: apple/Sources/Sargon/Util/SharedConstants.swift
// MARK: - SharedConstants
public enum SharedConstants {}

extension SharedConstants {
	public static let minRequiredXrdForAccountDeletion = constantMinRequiredXrdForAccountDeletion()
}

extension Account {
	public static let nameMaxLength = Int(constantEntityNameMaxLength())
}

extension Persona {
	public static let nameMaxLength = Int(constantEntityNameMaxLength())
}

extension DisplayName {
	public static let maxLength = Int(constantDisplayNameMaxLength())
}

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/SharedConstants.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.Account
import com.radixdlt.sargon.DisplayName
import com.radixdlt.sargon.Persona
import com.radixdlt.sargon.annotation.KoverIgnore
import com.radixdlt.sargon.constantDisplayNameMaxLength
import com.radixdlt.sargon.constantEntityNameMaxLength
import com.radixdlt.sargon.constantMinRequiredXrdForAccountDeletion
import com.radixdlt.sargon.extensions.SharedConstants.displayNameMaxLength
import com.radixdlt.sargon.extensions.SharedConstants.entityNameMaxLength

@KoverIgnore
object SharedConstants {
    val minRequiredXrdForAccountDeletion = constantMinRequiredXrdForAccountDeletion()
    val entityNameMaxLength = constantEntityNameMaxLength().toLong()
    val displayNameMaxLength = constantDisplayNameMaxLength().toLong()
}

@KoverIgnore
val Account.Companion.nameMaxLength: Long
    @KoverIgnore
    get() = entityNameMaxLength

@KoverIgnore
val Persona.Companion.nameMaxLength: Long
    @KoverIgnore
    get() = entityNameMaxLength

@KoverIgnore
val DisplayName.Companion.maxLength: Long
    @KoverIgnore
    get() = displayNameMaxLength